msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AcquireTokenSilentSupplier.java

```diff
@@ -53,29 +55,9 @@ AuthenticationResult execute() throws Exception {
                 clientApplication.serviceBundle().getServerSideTelemetry().incrementSilentSuccessfulCount();
             }
 
-            //Determine if the current token needs to be refreshed according to the refresh_in value
```

Comment:
nit: control flow via exception should be avoided (see line 48). 

https://stackoverflow.com/questions/729379/why-not-use-exceptions-as-regular-flow-of-control